data_processors/pipeline/orchestration/dragen_wts_step.py

```diff
@@ -26,90 +26,99 @@
 def perform(this_workflow: Workflow):
     logger.info(f"Preparing {WorkflowType.DRAGEN_WTS.value} workflows")
 
-    batcher = Batcher(
-        workflow=this_workflow,
-        run_step=WorkflowType.DRAGEN_WTS.value,
-        batch_srv=batch_srv,
-        fastq_srv=fastq_srv,
-        logger=logger,
+    this_sqr = this_workflow.sequence_run
+
+    # Check if all other DRAGEN_WTS_QC workflows for this run have finished
+    # If yes we continue to the DRAGEN WTS Workflow
+    # If not we wait (until all DRAGEN_WTS_QC worklfows for this run have finished
+    running: List[Workflow] = workflow_srv.get_running_by_sequence_run(
+        sequence_run=this_sqr,
+        workflow_type=WorkflowType.DRAGEN_WTS_QC
+    )
+    succeeded: List[Workflow] = workflow_srv.get_succeeded_by_sequence_run(
+        sequence_run=this_sqr,
+        workflow_type=WorkflowType.DRAGEN_WTS_QC
     )
 
-    if batcher.batch_run is None:
-        return batcher.get_skip_message()
+    subjects = list()
+    submitting_subjects = list()
+    if len(running) == 0:
+        logger.info("All QC workflows finished, proceeding to dragen wts preparation")
+
+        meta_list, run_libraries = metadata_srv.get_wts_metadata_by_wts_qc_runs(succeeded)
+        if not meta_list:
+            logger.warning(f"No dragen wts metadata found for given run libraries: {run_libraries}")
 
-    # prepare job list and dispatch to job queue
-    job_list = prepare_dragen_wts_jobs(batcher)
-    if job_list:
-        libsqs.dispatch_jobs(
-            queue_arn=libssm.get_ssm_param(SQS_DRAGEN_WTS_QUEUE_ARN),
-            job_list=job_list
-        )
+        job_list, subjects, submitting_subjects = prepare_dragen_wts_jobs(meta_list)
+
+        if job_list:
+            logger.info(f"Submitting {len(job_list)} WTS jobs for {submitting_subjects}.")
+            queue_arn = libssm.get_ssm_param(SQS_DRAGEN_WTS_QUEUE_ARN)
+            libsqs.dispatch_jobs(queue_arn=queue_arn, job_list=job_list)
+        else:
+            logger.warning(f"Calling to prepare_tumor_normal_jobs() return empty list, no job to dispatch...")
     else:
-        batcher.reset_batch_run()  # reset running if job_list is empty
+        logger.warning(f"DRAGEN_WGS_QC workflow finished, but {len(running)} still running. Wait for them to finish...")
```

Comment:
typo in logger

```
DRAGEN_WTS_QC
```

Reply:
Resolved in 7f4f03c68c8146c4034f141bf454a5ff7b4f5321